JASP-R-Interface/jaspResults/R/writeImage.R

```diff
@@ -146,3 +138,48 @@ redrawPlotJaspResults <- function(rec_plot)
 
   suppressWarnings(grDevices::replayPlot(rec_plot))
 }
+
+.decodeplot <- function(x, ...) {
+  UseMethod(".decodeplot", x)
+}
+
+.decodeplot.JASPgraphsPlot <- function(x) {
+  for (i in seq_along(x$subplots))
+    x$subplots[[i]] <- .decodeplot(x$subplots[[i]], returnGrob = FALSE)
+  return(x)
+}
+
+.decodeplot.gg <- function(x, returnGrob = TRUE) {
+  # TODO: do not return a grid object!
```

Comment:
What is the consequence of returning a grid object here?

Reply:
Right now, there isn't any difference in terms of output because .decodeplot is always called before drawing and we save the encoded plot in the state. 

It's probably better to save the decoded plot (and only decode once), but we can't do that because then we get a grid object and we can't get options for plot editing from a grid object. 

I'd rather return a ggplot object, but I'm returning a grid object right now because I don't know exactly how to decode certain ggplot functions that draw text (e.g., what if people put the name of a column as an axis tick mark? Thats's possible and I just don't know a catch all decoding method).